src/test/integration/IntegrationDeployer.t.sol

```diff
@@ -0,0 +1,489 @@
+            TransparentUpgradeableProxy(payable(address(strategyManager))),
+            address(strategyManagerImplementation),
+            abi.encodeWithSelector(
+                StrategyManager.initialize.selector,
+                eigenLayerReputedMultisig, //initialOwner
+                eigenLayerReputedMultisig, //initial whitelister
+                pauserRegistry,
+                0 // initialPausedStatus
+            )
+        );
+        // Slasher
+        eigenLayerProxyAdmin.upgradeAndCall(
+            TransparentUpgradeableProxy(payable(address(slasher))),
+            address(slasherImplementation),
+            abi.encodeWithSelector(
+                Slasher.initialize.selector,
+                eigenLayerReputedMultisig,
+                pauserRegistry,
+                0 // initialPausedStatus
+            )
+        );
+        // EigenPodManager
+        eigenLayerProxyAdmin.upgradeAndCall(
+            TransparentUpgradeableProxy(payable(address(eigenPodManager))),
+            address(eigenPodManagerImplementation),
+            abi.encodeWithSelector(
+                EigenPodManager.initialize.selector,
+                type(uint256).max, // maxPods
+                address(beaconChainOracle),
+                eigenLayerReputedMultisig, // initialOwner
+                pauserRegistry,
+                0 // initialPausedStatus
+            )
+        );
+        // Delayed Withdrawal Router
+        uint256 withdrawalDelayBlocks = 7 days / 12 seconds;
+        eigenLayerProxyAdmin.upgradeAndCall(
+            TransparentUpgradeableProxy(payable(address(delayedWithdrawalRouter))),
+            address(delayedWithdrawalRouterImplementation),
+            abi.encodeWithSelector(
+                DelayedWithdrawalRouter.initialize.selector,
+                eigenLayerReputedMultisig, // initialOwner
+                pauserRegistry,
+                0, // initialPausedStatus
+                withdrawalDelayBlocks
+            )
+        );
+
+        // Create base strategy implementation and deploy a few strategies
+        baseStrategyImplementation = new StrategyBase(strategyManager);
+
+        _newStrategyAndToken("Strategy1Token", "str1", 10e50, address(this)); // initialSupply, owner
+        _newStrategyAndToken("Strategy2Token", "str2", 10e50, address(this)); // initialSupply, owner
+        _newStrategyAndToken("Strategy3Token", "str3", 10e50, address(this)); // initialSupply, owner
+
+        ethStrats.push(BEACONCHAIN_ETH_STRAT);
+        mixedStrats.push(BEACONCHAIN_ETH_STRAT);
+
+        global = new Global();
+    }
+
+    /// @dev Deploy a strategy and its underlying token, push to global lists of tokens/strategies, and whitelist
+    /// strategy in strategyManager
+    function _newStrategyAndToken(string memory tokenName, string memory tokenSymbol, uint initialSupply, address owner) internal {
+        IERC20 underlyingToken = new ERC20PresetFixedSupply(tokenName, tokenSymbol, initialSupply, owner); 
```

Comment:
probably want to nix the tokenName and tokenSymbol inputs?
not sure about the 'owner' but maybe we should have a preset that makes some of these tests easier, like one that anyone can mint?

Reply:
i like these solely because i'm printing the token allocations in `_newUser`, in case a test fails and we want to see what sort of users we were generating

Reply:
"anyone can mint" -- actually apparently `StdCheats.deal` just takes care of the minting question? although in retrospect that may be because we're calling as the owner? i have no idea tbh, i grabbed it from yash's tests

Reply:
`StdCheats.deal` works by modifying storage manually :)